golden/example_test.go

```diff
@@ -1,11 +1,15 @@
-package golden
+package golden_test
 
-var t = &FakeT{}
+import (
+	"testing"
+
+	"github.com/gotestyourself/gotestyourself/golden"
+)
 
 func ExampleAssert() {
-	Assert(t, "foo", "foo-content.golden")
+	golden.Assert(&testing.T{}, "foo", "foo-content.golden")
```

Comment:
I think `&testing.T{}` might be confusing and further from the actual usage, since normally you just receive a `t`. We could use a `var t = &testing.T{}` at the package level instead of the fakeT.